service/controller/resource/deployment/templates.go

```diff
@@ -0,0 +1,1523 @@
+          },
+          "prefix":{
+            "value":"api"
+          },
+          "ipAddress":{
+            "value":"[reference('api_load_balancer_setup').outputs.ipAddress.value]"
+          }
+        }
+      }
+    },
+    {
+      "apiVersion":"2016-09-01",
+      "name":"kubernetes_etcd_dns_setup",
+      "type":"Microsoft.Resources/deployments",
+      "properties":{
+        "expressionEvaluationOptions": {
+          "scope": "inner"
+        },
+        "mode":"incremental",
+        "template": %s,
+        "parameters":{
+          "dnsZone":{
+            "value":"[concat(parameters('clusterID'), '.k8s.', parameters('dnsZones').api.name)]"
+          },
+          "GiantSwarmTags":{
+            "value":"[parameters('GiantSwarmTags')]"
+          },
+          "prefix":{
+            "value":"etcd"
+          },
+          "ipAddress":{
+            "value":"[reference('etcd_load_balancer_setup').outputs.ipAddress.value]"
+          }
+        }
+      }
+    },
+    {
+      "apiVersion":"2016-09-01",
+      "name":"container_setup",
+      "type":"Microsoft.Resources/deployments",
+      "properties":{
+        "expressionEvaluationOptions": {
+          "scope": "inner"
+        },
+        "mode":"incremental",
+        "template": %s,
+        "parameters":{
+          "blobContainerName":{
+            "value":"[parameters('blobContainerName')]"
+          },
+          "clusterID":{
+            "value":"[parameters('clusterID')]"
+          },
+          "GiantSwarmTags":{
+            "value":"[parameters('GiantSwarmTags')]"
+          },
+          "storageAccountName":{
+            "value":"[parameters('storageAccountName')]"
+          }
+        }
+      }
+    },
+    {
+      "apiVersion":"2016-09-01",
+      "name":"kubernetes_ingress_wildcard_dns_setup",
+      "type":"Microsoft.Resources/deployments",
+      "properties":{
+        "expressionEvaluationOptions": {
+          "scope": "inner"
+        },
+        "mode":"incremental",
+        "template": %s,
+        "parameters":{
+          "dnsZone":{
+            "value":"[concat(parameters('clusterID'), '.k8s.', parameters('dnsZones').ingress.name)]"
+          },
+          "GiantSwarmTags":{
+            "value":"[parameters('GiantSwarmTags')]"
+          },
+          "prefix":{
+            "value":"*"
+          },
+          "cname":{
+            "value":"[concat('ingress.', parameters('clusterID'), '.k8s.', parameters('dnsZones').ingress.name)]"
+          }
+        }
+      }
+    }
+  ]
+}`
+
+func getARMTemplate() (map[string]interface{}, error) {
+	contents := make(map[string]interface{})
+	if err := json.Unmarshal([]byte(getARMTemplateAsString()), &contents); err != nil {
+		return nil, err
+	}
+	return contents, nil
+}
+
+func getARMTemplateAsString() string {
+	return fmt.Sprintf(main, securityGroups, routeTable, virtualNetwork, publicLoadBalancer, privateLoadBalancer, kubernetesLoadBalanacer, dnsA, dnsA, containerSetup, dnsCname)
```

Comment:
Here I build the whole template.

Reply:
Well, of course it was better to have actual json files, expecially from a code completion/IDE point of view.
Not a blocker but maybe in the future we could check solutions like https://github.com/markbates/pkger ?

Reply:
If we get go modules working we can try that, I think it will be a better solution.